python/graphstorm/dataloading/dataset.py

```diff
@@ -579,10 +579,26 @@ def _get_node_set(self, ntypes, mask):
                 idx = [] if idx is None else idx
                 num_data += len(idx)
                 # If there are validation/test data globally, we should add them to the dict.
-                if dist_sum(len(idx)) > 0:
-                    idxs[ntype] = idx
-
-                logging.debug('part %d | ntype %s, mask %s | num nodes: %d',
+                total_num_idx = dist_sum(len(idx))
+                if total_num_idx > 0:
+                    if total_num_idx >= get_world_size():
+                        # The size of the validation or test set is larger
+                        # than the world size. Each validation/test dataloader
+                        # will not be empty
+                        idxs[ntype] = idx
+                    else:
+                        # There is not enough validation or test data.
+                        # One or more validation/test dataloader will be
+                        # empty, which will cause an evaluation error.
+                        #
+                        # To avoid the error, force each trainer or
+                        # inferencer to use the entire validation
+                        # or test set.
+                        idx = th.nonzero(g.nodes[ntype].data[msk][ \
```

Comment:
Shall we add a log here to indicate users that we are using the entire validation/test set? We can set it to debug message.

Reply:
Good callout